docs/workflow/testing/mono/testing.md

```diff
@@ -1,54 +1,114 @@
+```
+cd ../mono/netcore
+make run-tests-coreclr CoreClrTest="bash ../../artifacts/tests/coreclr/OSX.x64.Release/JIT/opt/InstructionCombining/DivToMul/DivToMul.sh"
+```
+
+ * Running all runtime tests:
+```
+cd ../mono/netcore
+make run-tests-coreclr-all
+```
+
+### Testing on WebAssembly:
+```
+$(REPO_ROOT)/src/tests/build.sh -skipstressdependencies -excludemonofailures os Browser wasm <Release/Debug>
+```
+
+The last few lines of the build log should contain something like this:
+```
+--------------------------------------------------
+ Example run.sh command
+
+ src/tests/run.sh --coreOverlayDir=<repo_root>artifacts/tests/coreclr/Browser.wasm.Release/Tests/Core_Root --testNativeBinDir=<repo_root>/artifacts/obj/coreclr/Browser.wasm.Release/tests --testRootDir=<repo_root>/artifacts/tests/coreclr/Browser.wasm.Release --copyNativeTestBin Release
+--------------------------------------------------
 ```
-dotnet build /t:RunCoreClrTests $(REPO_ROOT)/src/mono/mono.proj
+
+Run that command, adding `wasm` to the end.
+
+### Testing on Android x64:
+```
+$(REPO_ROOT)/src/tests/build.sh -skipstressdependencies -excludemonofailures os Android x64 <Release/Debug>
+```
+
+The last few lines of the build log should contain something like this:
+```
+--------------------------------------------------
+ Example run.sh command
+
+ src/tests/run.sh --coreOverlayDir=<repo_root>artifacts/tests/coreclr/Android.x64.Release/Tests/Core_Root --testNativeBinDir=<repo_root>/artifacts/obj/coreclr/Android.x64.Release/tests --testRootDir=<repo_root>/artifacts/tests/coreclr/Android.x64.Release --copyNativeTestBin Release
+--------------------------------------------------
 ```
+Run that command, adding `Android` at the end.
+
+For more details about internals of the runtime tests, please refer to the [CoreCLR testing documents](../coreclr)
 
-If you want to run individual tests, execute this command:
+## Libraries tests
+* Build and run library tests against Mono JIT:
+    * cd into the test library of your choice (`cd $(REPO_ROOT)/src/libraries/<library>/tests`)
+    * Run the tests
 
 ```
-dotnet build /t:RunCoreClrTest /p:CoreClrTest="<TestName>" $(REPO_ROOT)/src/mono/mono.proj
+$(REPO_ROOT)/dotnet.sh build /t:Test /p:RuntimeFlavor=mono /p:Configuration=<Release/Debug>
 ```
 
-## Running Library Tests
-Running library tests against Mono is straightforward regardless of configuration.  Simply run the following commands:
+* Build and run library tests against Webassembly, Android or iOS. See instructions located in [Library testing document folder](../libraries/)
 
-1. Build and set the RuntimeFlavor
+# Running the Mono samples
+There are a few convenient samples located in `$(REPO_ROOT)/src/mono/netcore/sample`, which could help you test your program easily with different flavors of Mono or do a sanity check on the build.
+
+## Sample for desktop Mono
+It lives in `HelloWorld` folder. 
+
+To run that program, you could simply cd to that directory and execute
```

Comment:
```suggestion
To run the sample, cd to the directory and execute:
```
Indirect language. Also, avoid "simply" unless necessary in documentation, for the same reasons as words like "obviously": it rarely adds clarity and can potentially be insulting to newer contributors.